react-renderer/src/inspector/views/mod/LabeledInput.tsx

```diff
@@ -0,0 +1,258 @@
+        this.updateAttr(event.target.id, teval); // a little hacky - id is name of property
+    }
+    // when an input corresponding to a part of an attribute is modified instead of an entire attribute
+    public handleSubChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+        const ptinfo = event.target.id.split("_");
+        const xory = (ptinfo[3] === "x" ? 0 : 1);
+        const teval = (isNaN(Number(event.target.value))) ? event.target.value : +event.target.value;
+        const subpaths = _.cloneDeep(this.props.eValue.contents);
+        const subpath = subpaths[+ptinfo[1]];   // get specific subpath
+        const pt = subpath.contents[+ptinfo[4]];    // get specific point
+        pt.contents[xory] = teval;
+        this.updateAttr(ptinfo[2], subpaths);
+    }
+    public handleCheck = (event: React.ChangeEvent<HTMLInputElement>) => {
+        this.updateAttr(event.target.id, event.target.checked);
+    }
+    public keyDown = (event: React.KeyboardEvent<HTMLInputElement>) => {
+        if (event.keyCode === 13) {
+            const etgt = (event.target) as HTMLInputElement;
+            const teval = (isNaN(Number(etgt.value))) ? etgt.value : +etgt.value; // convert to number if necessary
+            this.updateAttr(etgt.id, teval);
+        }
+    }
+    public handleColor = (event: React.ChangeEvent<HTMLInputElement>) => {
+        const hex = event.target.value;
+        this.updateAttr(event.target.id, {
+            tag: "RGBA",
+            contents: this.toRGBA(hex)
+        })
+    }
+    // https://stackoverflow.com/questions/21646738/convert-hex-to-rgba
+    public toRGBA = (hex: string) => {
+        const r = parseInt(hex.slice(1, 3), 16) / 255,
+            g = parseInt(hex.slice(3, 5), 16) / 255,
+            b = parseInt(hex.slice(5, 7), 16) / 255;
+        return ([r, g, b, this.props.eValue.contents[3]] as [number, number, number, number])
+        
+        // this will destroy alpha specified in style.
+        // todo - pass over alpha from style specification? either way color picker prevents alpha styling
+        // maybe there is a custom color picker somewhere that uses alpha... 
+    }
+    // todo - maybe make toCanvas definable in JSON? range doesn't cover all possible use cases for toCanvas
+    public makeRange = () => {  
+        const {inputProps, eAttr} = this.props;
+        return (<input id={eAttr} type= "range" onChange={this.handleChange} min= {toCanvas(inputProps.min!)} max={toCanvas(inputProps.max!)
+        } value={_.round(this.props.eValue.contents)} />)
+    }
+    // for use in makePointRange or potentially other multi-input attributes (would need to be modified)
+    public makeSublabel = (id: string) => {
+        const itemprops = id.split("_");
+        const subindex = itemprops[4];
+        const ptindex = itemprops[3] === "x" ? 0 : 1;
+        console.log(this.state.eValue.contents[+itemprops[1]].contents[+subindex].contents[ptindex]);
```

Comment:
Remove extraneous `console.log`s?

Reply:
Addressed in 2b9315a4a27f35fb751ac317f15cdbf9a18ec5e4